### PR TITLE
Fix `F.softmax` and `F.log_softmax` with `axis=-1` on gpu

### PR DESCRIPTION
**Pull request by toslunar**
This PR fixes a bug in #5381.

`_get_tensor4d_shape` implicitly assumed `axis` is nonnegative.  (`shape[slice(axis + 1, len(shape))]` doesn't work if `axis=-1`.)

### CODE REVIEW

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by kmaehashi** [comment]
LGTM! Test failures are not related to this PR.
